### PR TITLE
virtio block cache mode support

### PR DESCRIPTION
**Pull request by conghuic23**
dm: storage: support writethru and writeback mode
doc: update virtio-blk usage in HLD
dm: virtio: remove hv_caps from virtio_ops
dm: storage: support cache mode toggling

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: doc/developer-guides/virtio-hld.rst [hunk 1]
```diff
@@ -530,7 +530,25 @@ VTBLK_F_TOPOLOGY(bit 10)
...
-   -s <slot>,virtio-blk,<filepath>
+   -s <slot>,virtio-blk,<filepath>[,options]
+   
+   options:
+
+   writethru: write operation is reported completed only when the
+   data has been written to physical storage.
+
+   writeback: write operation is reported completed when data is
+   placed in page cache. Need to be flushed to the physical storage.
```
Comment (dbkinder) [inline doc/developer-guides/virtio-hld.rst]: change ``Need`` to ``Needs``

File: doc/developer-guides/virtio-hld.rst [hunk 2]
```diff
@@ -530,7 +530,25 @@ VTBLK_F_TOPOLOGY(bit 10)
...
+
+   ro: open file with readonly mode.
+
+   sectorsize:
+   1> sectorsize=<sector size>/<physical sector size>
+   2> sectorsize=<sector size>
+   default value for sector size and physical sector size are 512
```
Comment (dbkinder) [inline doc/developer-guides/virtio-hld.rst]: change ``value`` to ``values``

File: doc/developer-guides/virtio-hld.rst (hunk 1 above)
Comment (conghuic23) [reply doc/developer-guides/virtio-hld.rst]: Fixed

File: doc/developer-guides/virtio-hld.rst (hunk 2 above)
Comment (conghuic23) [reply doc/developer-guides/virtio-hld.rst]: Fixed

**Comment by lijinxia** [comment]
OK to verify

**Pull request merged**